docs/docs/aleph_zero/sdk.md

```diff
@@ -257,24 +236,101 @@ const sqrtPriceLimit = calculateSqrtPriceAfterSlippage(
+  liquidityGross: Liquidity
+  sqrtPrice: SqrtPrice
+  feeGrowthOutsideX: FeeGrowth
+  feeGrowthOutsideY: FeeGrowth
+  secondsOutside: bigint
+}
+
+const tickState: Tick = await invariant.getTick(signer, poolKey, tickIndex)
+```
+
+- Get Pool
+
+```typescript
+interface Pool {
+  liquidity: Liquidity
+  sqrtPrice: SqrtPrice
+  currentTickIndex: bigint
+  feeGrowthGlobalX: FeeGrowth
+  feeGrowthGlobalY: FeeGrowth
+  feeProtocolTokenX: TokenAmount
+  feeProtocolTokenY: TokenAmount
+  startTimestamp: bigint
+  lastTimestamp: bigint
+  feeReceiver: string
+}
+
+const poolState: Pool = await invariant.getPool(signer, TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
+```
+
+- Get Pools
+
+```typescript
+const pools: Pool[] = await invariant.getPools(signer)
+```
 
 :::note Position indexing
-Remember that positions are indexed from 0. So if you create position, its id will be 0 and your next positions id will be 1.
+Remember that positions are indexed from 0. So if you create position, its index will be 0 and your next positions index will be 1.
 :::
 
+- Get Position
+
+```typescript
+interface Position {
+  poolKey: PoolKey
+  liquidity: Liquidity
+  lowerTickIndex: bigint
+  upperTickIndex: bigint
+  feeGrowthInsideX: FeeGrowth
+  feeGrowthInsideY: FeeGrowth
+  lastBlockNumber: bigint
+  tokensOwedX: TokenAmount
+  tokensOwedY: TokenAmount
+}
+
+const positionState: Position = await invariant.getPosition(signer, owner.address, positionIndex)
+```
+
+- Get Positions
+
+```typescript
+const positions: Position[] = await invariant.getPositions(
+  await invariant.getPositions(signer, owner.address)
+)
+```
+
+### Query states and Calculate Fee
+
+To query the state and calculate fees within the system, several functions are utilized. Positions, ticks, and pools are accessed to gather information about the state, and the calculateFee function is used to determine the amount of unclaimed tokens. The process is described below:
+
 ```typescript
-// --snip--
+// -- snip --
 
 // query state
-const poolAfter = await invariant.getPool(account, TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
-// last parameter here is position id, positions are indexed from 0
-const positionAfter = await invariant.getPosition(account, account.address, 0n)
-const lowerTickAfter = await invariant.getTick(account, poolKey, positionAfter.lowerTickIndex)
-const upperTickAfter = await invariant.getTick(account, poolKey, positionAfter.upperTickIndex)
+const poolAfter: Pool = await invariant.getPool(account, TOKEN0_ADDRESS, TOKEN1_ADDRESS, feeTier)
+const positionAfter: Position = await invariant.getPosition(account, account.address, 0n)
+const lowerTickAfter: Tick = await invariant.getTick(account, poolKey, positionAfter.lowerTickIndex)
+const upperTickAfter: Tick = await invariant.getTick(account, poolKey, positionAfter.upperTickIndex)
 
 // pools, ticks and positions have many fee growth fields that are used to calculate fees,
 // by doing that off chain we can save gas fees,
```

Comment:
To removed this: "by doing that off chain we can save gas fees ...".
We simply want to calculate it without signing and sending transaction. Consider the case where someone has to sign transaction using hardware wallet just to check how much fee has already accumulated